frontend/app/src/components/hexagonalBoard/HexagonalBoard.tsx

```diff
@@ -0,0 +1,117 @@
+import {
+	useState,
+	ReactNode,
+	useEffect,
+	forwardRef,
+	useImperativeHandle,
+} from "react";
+import HexagonalTile, {
+	HexagonalTileProps,
+} from "@/components/hexagonalTile/HexagonalTile.tsx";
+import "./HexagonalBoard.css";
+
+const defaultTileProps: HexagonalTileProps = {
+	fill: "",
+	isDimmed: false,
+	stroke: "gray",
+	strokeWidth: "16",
+	identifier: "",
+	textContent: "",
+};
+
+interface HexagonalBoardProps {
+	rows: number;
+	tilesPerRow: number;
+	ref?: React.Ref<{
+		updateTile: (
+			row: number,
+			col: number,
+			props: Partial<HexagonalTileProps>,
+		) => void;
+	}>;
+}
+
+const HexagonalBoard = forwardRef(function HexagonalBoard(
```

Comment:
maybe this'd be quicker in person or over discord sometime, but I'd love to know why `forwardRef` is needed and what exactly it does in this case

Reply:
In short: normally you can only create functions in higher components and pass them down to child components, by using a forwardRef you can (kinda) do the opposite: declare a function in the child component and expose them to parent components.